src/anndata/_io/specs/lazy_methods.py

```diff
@@ -154,11 +173,104 @@ def read_h5_array(
     return da.map_blocks(make_chunk, dtype=dtype, chunks=chunk_layout)
 
 
+@_LAZY_REGISTRY.register_read(ZarrArray, IOSpec("string-array", "0.2.0"))
 @_LAZY_REGISTRY.register_read(ZarrArray, IOSpec("array", "0.2.0"))
 def read_zarr_array(
-    elem: ZarrArray, *, _reader: DaskReader, chunks: tuple[int, ...] | None = None
+    elem: ZarrArray, *, _reader: LazyReader, chunks: tuple[int, ...] | None = None
 ) -> DaskArray:
     chunks: tuple[int, ...] = chunks if chunks is not None else elem.chunks
     import dask.array as da
 
     return da.from_zarr(elem, chunks=chunks)
+
+
+@_LAZY_REGISTRY.register_read(ZarrGroup, IOSpec("dataframe", "0.2.0"))
+@_LAZY_REGISTRY.register_read(H5Group, IOSpec("dataframe", "0.2.0"))
+def read_dataframe(
+    elem: H5Group | ZarrGroup,
+    *,
+    _reader: LazyReader,
+    chunks: tuple[int, ...] | None = None,
+) -> Dataset2D:
+    from anndata.experimental.backed._compat import xr
+    from anndata.experimental.backed._lazy_arrays import CategoricalArray, MaskedArray
+    from anndata.experimental.backed._xarray import Dataset2D
+
+    iter_object = [(k, elem[k]) for k in elem.attrs["column-order"]] + [
+        (elem.attrs["_index"], elem[elem.attrs["_index"]])
+    ]
+    d = {k: _reader.read_elem(v) for k, v in iter_object}
+    d_with_xr = {}
+    elem_name = get_elem_name(elem)
+    index_label = f'{elem_name.replace("/", "")}_names'
+    index = d[elem.attrs["_index"]]  # no sense in reading this in multiple times
+    for k in d:
+        v = d[k]
+        if isinstance(v, DaskArray) and k != elem.attrs["_index"]:
+            d_with_xr[k] = xr.DataArray(v, coords=[index], dims=[index_label], name=k)
+        elif (
+            isinstance(v, CategoricalArray) or isinstance(v, MaskedArray)
+        ) and k != elem.attrs["_index"]:
+            variable = xr.Variable(
+                data=xr.core.indexing.LazilyIndexedArray(v), dims=[index_label]
+            )
+            d_with_xr[k] = xr.DataArray(
+                variable,
+                coords=[index],
+                dims=[index_label],
+                name=k,
+            )
+        elif k == elem.attrs["_index"]:
+            d_with_xr[index_label] = xr.DataArray(
+                v, coords=[v], dims=[index_label], name=index_label
+            )
+        else:
+            d_with_xr[k] = v
+    return Dataset2D(d_with_xr)
+
+
+@_LAZY_REGISTRY.register_read(ZarrGroup, IOSpec("categorical", "0.2.0"))
+@_LAZY_REGISTRY.register_read(H5Group, IOSpec("categorical", "0.2.0"))
+def read_categorical(
```

Comment:
It's part of the required signature from `read_elem_as_dask`.  I can make the `LazyReader`/`DaskReader` class check for the `chunks` param and use it if it is needed and not otherwise.  So we could drop it here.